sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1876,20 +1876,12 @@ def insert(
         )
         snapshot = kwargs["snapshot"]
         snapshots = kwargs["snapshots"]
+
         if (
-            (
-                isinstance(query_or_df, exp.Expression)
-                and snapshot.is_materialized_view
-                and deployability_index.is_deployable(snapshot)
-                and model.render_query(
-                    snapshots=snapshots,
-                    deployability_index=deployability_index,
-                    engine_adapter=self.adapter,
-                )
-                == query_or_df
-            )
-            or self.adapter.HAS_VIEW_BINDING
-        ) and self.adapter.table_exists(table_name):
+            not snapshot.is_materialized_view
```

Comment:
Previously we were skipping *only* for materialized views and never for regular ones. This changes it. I think we should remove the whole skipping logic up to and including line 1886.